src/core/jpg.js

```diff
@@ -163,12 +163,16 @@ var JpegImage = (function JpegImageClosure() {
           } else if (nextByte === /* EOI = */ 0xd9) {
             if (parseDNLMarker) {
               // NOTE: only 8-bit JPEG images are supported in this decoder.
-              const maybeScanLines = blockRow * 8;
+              const maybeScanLines = blockRow * (frame.precision === 8 ? 8 : 0);
```

Comment:
Not completely related to the patch, but a better-safe-than-sorry sort of change when I had to touch related code anyway.